src/libtls/tls_crypto.c

```diff
@@ -1554,12 +1584,46 @@ CALLBACK(group_filter, bool,
 	return FALSE;
 }
 
+CALLBACK(config_filter, bool,
+	void *null, enumerator_t *orig, va_list args)
+{
+	diffie_hellman_group_t group, *out;
+	tls_named_group_t *curve;
+	char *plugin;
+	int i;
+
+	VA_ARGS_VGET(args, out, curve);
+
+	while (orig->enumerate(orig, &group, &plugin))
+	{
+		for (i = 0; i < countof(curves); i++)
+		{
+			if (curves[i].group == group &&
+				filter_curve_config(tls_ec_group_to_curve(curves[i].group)))
+			{
+				if (out)
+				{
+					*out = curves[i].group;
+				}
+				if (curve)
+				{
+					*curve = curves[i].curve;
+				}
+				return TRUE;
+			}
+		}
+	}
+	return FALSE;
+}
+
 METHOD(tls_crypto_t, create_ec_enumerator, enumerator_t*,
 	private_tls_crypto_t *this)
 {
 	return enumerator_create_filter(
-							lib->crypto->create_dh_enumerator(lib->crypto),
-							group_filter, NULL, NULL);
+							enumerator_create_filter(
+								lib->crypto->create_dh_enumerator(lib->crypto),
+								group_filter, NULL, NULL),
+							config_filter, NULL, NULL);
```

Comment:
I spilt the two filters as you mentioned, but technically there is no difference in what `create_ec_enumerator` returns, is it? The callback `group_filter` is only used from within this method.